src/components/Objects/ProtectedRoute/ProtectedRoute.jsx

```diff
@@ -0,0 +1,27 @@
+import React from 'react';
+import { Route, Redirect } from 'react-router-dom';
+
+import { useUser } from '../../../contexts/UserContext';
+
+export default function ProtectedRoute({ component: Component, ...rest }) {
```

Comment:
I don't see this ProtectedRoute Component being used anywhere. Is this waiting to be used in the future?

Reply:
Correct. We would use this for all routes that require the user to be authenticated instead of the usual `<Route />` component that react-router-dom exports.